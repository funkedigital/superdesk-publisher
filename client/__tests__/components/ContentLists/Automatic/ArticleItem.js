import React from "react";
import ArticleItem from "../../../../components/ContentLists/Automatic/ArticleItem";
import { render } from "@testing-library/react";
import moment from 'moment';

jest.mock("moment", () => () => ({ fromNow: () => "2 days ago" }));

describe("ContentLists/Automatic/ArticleItem", () => {
  const item = {
    content: {
      title: "test",
      route: {
        name: "route"
      },
      authors: [{ name: "author" }],
      published_at: "2017-07-26T13:43:33+00:00",
      article_statistics: {
        page_views_number: 10,
        internal_click_rate: 5,
        impressions_number: 15
      }
    }
  };

  it("renders correctly", () => {
    const { container } = render(
      <ArticleItem
        item={item}
        style={{}}
        openPreview={jest.fn()}
        previewItem={{}}
        pinUnpin={jest.fn()}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
