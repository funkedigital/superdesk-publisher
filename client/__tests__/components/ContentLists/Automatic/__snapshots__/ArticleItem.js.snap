// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentLists/Automatic/ArticleItem renders correctly 1`] = `
<div
  class="sd-list-item sd-list-item--activated"
>
  <div
    class="sd-list-item__column sd-list-item__column--grow sd-list-item__column--no-border"
  >
    <div
      class="sd-list-item__row"
    >
      <span
        class="sd-overflow-ellipsis sd-list-item__text-strong"
      >
        test
      </span>
    </div>
    <div
      class="sd-list-item__row"
    >
      <span
        class="sd-overflow-ellipsis sd-list-item--element-grow"
      >
        2 days ago
      </span>
      <span
        class="label"
      >
        route
      </span>
    </div>
  </div>
  <div
    class="sd-list-item__action-menu sd-list-item__action-menu--direction-row"
  >
    <button
      class="pull-right"
      flow="left"
      sd-tooltip="Pin"
      title="Pin"
    >
      <i
        class="icon-pin"
      />
    </button>
    <button
      flow="left"
      sd-tooltip="remove"
      title="Remove"
    >
      <i
        class="icon-trash"
      />
    </button>
  </div>
</div>
`;
